### PR TITLE
NOTICK: Update the top-level README with a summary of each plugin.

### PR DESCRIPTION
**Pull request by chrisr3**
Provide a brief description of what each Gradle plugin actually _does_, in case anyone was wondering...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+declared to be "stable".
+
+- [`net.corda.plugins.jar-filter`](jar-filter/README.md):
+This plugin allows us to delete certain annotated classes, methods and
+fields from the compiled byte-code inside a jar file. It can also rewrite
+Kotlin classes' `@kotlin.Metadata` annotations to make them consistent 
+again with their edited byte-code. It has been successfully tested with
+Kotlin 1.2.x - 1.3.61.
+
+- [`net.corda.plugins.publish-utils`](publish-utils/README.rst): **_Here
```
Comment (JamesHR3) [inline README.md]: This is interesting. Is the implication that we should try and move away from this and deprecate it? Or does it do something useful but in a way that's scarily magical and therefore shouldn't be touched? I'd be interested to know what the dragons are!

File: README.md [hunk 2]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+This plugin provides `Cordform` and `Dockerform` Gradle tasks for creating
+test deployments of multiple Corda nodes and their CorDapps. It also invokes
+the Network Bootstapper over the deployment for you, and provides you with a
+`runnodes` script to boot it all up afterwards.
+
+- [`net.corda.plugins.quasar-utils`](quasar-utils/README.rst):
+This plugin configures a Gradle module to use Quasar. Specifically:
+    - It allows you to specify the Maven group, version and classifier of
+the `quasar-core` artifact to use.
+    - Adds the `quasar-core` artifact, along with all of its trensitive
```
Comment (blsemo) [inline README.md]: trensitive -> transitive

File: README.md [hunk 3]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+Kotlin classes' `@kotlin.Metadata` annotations to make them consistent 
+again with their edited byte-code. It has been successfully tested with
+Kotlin 1.2.x - 1.3.61.
+
+- [`net.corda.plugins.publish-utils`](publish-utils/README.rst): **_Here
+be Dragons!_** This plugin helps configure Gradle's `maven-publish` plugin
+to publish Corda modules to R3's Artifactory. However, its workings probably
+shouldn't be allowed and they should _most definitely_ **never** be copied.
+Unfortunately, my prior attempts at rewriting it also broke its interaction
+with JFrog's Artifactory plugin :-(.
```
Comment (blsemo) [inline README.md]: Can we reword this to be a tad more serious and dangerous sounding? Also, I really don't like emojis in docs.

File: README.md [hunk 4]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+This plugin packages your CorDapp classes into a single jar file, along with
+the contants of all of the jar's dependent jars. Dependencies which are added
+to Gradle's `cordaCompile`, `cordaRuntime` and `cordapp` configurations are
+excluded from this packaging. The final jar is then signed.
+
+- [`net.corda.plugins.cordformation`](cordformation/README.rst):
+This plugin provides `Cordform` and `Dockerform` Gradle tasks for creating
+test deployments of multiple Corda nodes and their CorDapps. It also invokes
+the Network Bootstapper over the deployment for you, and provides you with a
+`runnodes` script to boot it all up afterwards.
```
Comment (szymonsztuka) [inline README.md]: Maybe to add partiall task declaration exmaple and mention. it needs to depend on jar task, e.g.:

The task (by convetion named `deployNodes`) should depened on `jar` task,  e.g.:
``
task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['jar']) {
[...]
}
``

File: README.md [hunk 5]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+
+The plugins themselves fall into two categories: those intended for developing CorDapps, and those which are primarily used to build Corda itself.
+
+### Plugins for CorDapp development.
+
+- [`net.corda.plugins.cordapp`](cordapp/README.md):
+This plugin packages your CorDapp classes into a single jar file, along with
+the contants of all of the jar's dependent jars. Dependencies which are added
+to Gradle's `cordaCompile`, `cordaRuntime` and `cordapp` configurations are
+excluded from this packaging. The final jar is then signed.
```
Comment (szymonsztuka) [inline README.md]: What will happen if `cordapp` entry is omitted , will a JAR contain any default values for `name`/`vendor`/`targetPlatformVersion` etc?
In other words is this obligatory:
```
cordapp {
    info {
        name "Trader Demo"
        vendor "R3"
        targetPlatformVersion corda_platform_version.toInteger()
        minimumPlatformVersion 1
    }
}
```

File: README.md (hunk 1 above)
Comment (chrisr3) [reply README.md]: The `publish-utils` plugin is _definitely_ useful, but is also _**Really Bad Gradle Code!**_

Its Most Evil feature is that setting the `name` field in the `publish` block _executes code_:
```gradle
publish {
    name jar.baseName
}
```
which means that anything configured _after_ `name` here will be ignored!

File: README.md [hunk 6]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+These plugins are unlikely to be useful to CorDapp developers outside of R3.
+
+- [`net.corda.plugins.api-scanner`](api-scanner/README.md):
+This plugin scans the `public` and `protected` classes inside a Gradle
+module's "primary" jar artifact and writes a summary of their `public`
+and `protected` methods and fields into an output file. The "primary"
+jar is assumed by default to be the one without an `archiveClassifier`,
+although this is configurable. Its goal is to alert Corda developers to
+accidental breaks in our public ABI for those Corda modules we have
+declared to be "stable".
```
Comment (szymonsztuka) [inline README.md]: Maybe worth metioning, that the plugin is run in our CI build pipeline?

File: README.md [hunk 7]
```diff
@@ -31,3 +31,60 @@ The version number of the "bleeding edge" in `master` is always a `-SNAPSHOT` ve
...
+
+- [`net.corda.plugins.jar-filter`](jar-filter/README.md):
+This plugin allows us to delete certain annotated classes, methods and
+fields from the compiled byte-code inside a jar file. It can also rewrite
+Kotlin classes' `@kotlin.Metadata` annotations to make them consistent 
+again with their edited byte-code. It has been successfully tested with
+Kotlin 1.2.x - 1.3.61.
```
Comment (szymonsztuka) [inline README.md]: When or who uses this pluging? E.g. 'Plugin is used for developing DJVM' (?).

File: README.md (hunk 5 above)
Comment (szymonsztuka) [reply README.md]: Also maybe point out: 
The detailed documention of ``cordapp`` and ``cordformation`` plugins is available on https://docs.corda.net/cordapp-build-systems.html. (?)

File: README.md (hunk 4 above)
Comment (chrisr3) [reply README.md]: This is _supposed_ to be a brief summary only. I do also provide links to other documentation for each plugin.

File: README.md (hunk 5 above)
Comment (chrisr3) [reply README.md]: This is already mentioned by [the `cordapp` plugin's own README](cordapp/README.md).
